### PR TITLE
Fix/1.0.x/chat 160

### PR DESCRIPTION
**Pull request by rdenarie**
This PR aim to localize meeting notes sent by mail or saved in wiki

### CODE REVIEW

**Comment by trangvh** [comment]
For resource bundle files, you only need to keep Resource.properties and Resource_en.properties. The others will be created by Crowdin plugin.

**Comment by rdenarie** [comment]
Hi Trang,
I add a new commit removing unecessary properties files.

File: server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource.properties [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+exoplatform.chat.meetingnotes=Meeting Notes
+
+exoplatform.chat.meetingnotes.questions=Questions
+exoplatform.chat.meetingnotes.question=Question
+exoplatform.chat.meetingnotes.links=Links
+exoplatform.chat.meetingnotes.link=Link
+exoplatform.chat.meetingnotes.files=Files
+exoplatform.chat.meetingnotes.file=File
+exoplatform.chat.meetingnotes.raiseHand=
```
Comment (trangvh) [inline server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource.properties]: Key without value. It should be "Raise the hand"

File: server/src/main/java/org/exoplatform/chat/model/ReportBean.java [hunk 2]
```diff
@@ -145,7 +146,7 @@ else if ("type-task".equals(options.get("type").toString()))
             task.setDueDate(options.get("dueDate").toString());
             task.setUsername(options.get("username").toString());
             addTask(task);
-            msg = "[ Task \""+options.get("task").toString()+"\" assigned to "+options.get("fullname").toString()+" - due "+options.get("dueDate").toString()+" ]";
+            msg = "[ "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.task", locale)+" \""+options.get("task").toString()+"\" "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.task.assignedTo", locale)+" "+options.get("fullname").toString()+" - "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.task.due", locale)+" "+options.get("dueDate").toString()+" ]";
```
Comment (trangvh) [inline server/src/main/java/org/exoplatform/chat/model/ReportBean.java]: These labels should be inside a unique sentence for translation.

File: server/src/main/java/org/exoplatform/chat/model/ReportBean.java [hunk 3]
```diff
@@ -156,19 +157,19 @@ else if ("type-event".equals(options.get("type").toString()))
             event.setEndDate(options.get("endDate").toString());
             event.setEndTime(options.get("endTime").toString());
             addEvent(event);
-            msg = "[ Event \""+options.get("summary").toString()+"\" from "+options.get("startDate").toString()+" "+options.get("startTime").toString()+" to "+options.get("endDate").toString()+" "+options.get("endTime").toString()+" ]";
+            msg = "[ "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.event", locale)+" \""+options.get("summary").toString()+"\" "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.event.from", locale)+" "+options.get("startDate").toString()+" "+options.get("startTime").toString()+" "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.event.to", locale)+" "+options.get("endDate").toString()+" "+options.get("endTime").toString()+" ]";
```
Comment (trangvh) [inline server/src/main/java/org/exoplatform/chat/model/ReportBean.java]: Same comment as Task.

File: server/src/main/java/org/exoplatform/chat/model/ReportBean.java [hunk 4]
```diff
@@ -156,19 +157,19 @@ else if ("type-event".equals(options.get("type").toString()))
...
           }
           else if ("call-join".equals(options.get("type").toString()))
           {
-            msg = "[ Joined meeting ]";
+            msg = "[ "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.joined", locale)+" ]";
           }
           else if ("call-on".equals(options.get("type").toString()))
           {
-            msg = "[ Meeting started ]";
+            msg = "[ "+ ChatUtils.appRes("exoplatform.chat.meetingnotes.finished", locale)+" ]";
```
Comment (trangvh) [inline server/src/main/java/org/exoplatform/chat/model/ReportBean.java]: Swap lines 168 and 172.

File: server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource_en.properties [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+exoplatform.chat.meetingnotes.tasks.assignee=Assignee
+exoplatform.chat.meetingnotes.tasks.due=Due
+exoplatform.chat.meetingnotes.task=Task
+exoplatform.chat.meetingnotes.task.assignedTo=assigned to
+exoplatform.chat.meetingnotes.task.due=due
+
+exoplatform.chat.meetingnotes.events=Events
+exoplatform.chat.meetingnotes.events.event=Event
+exoplatform.chat.meetingnotes.events.start=Start
+exoplatform.chat.meetingnotes.events.end=EZnd
```
Comment (trangvh) [inline server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource_en.properties]: EZnd -> End

File: server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource.properties [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+exoplatform.chat.meetingnotes.tasks.assignee=Assignee
+exoplatform.chat.meetingnotes.tasks.due=Due
+exoplatform.chat.meetingnotes.task=Task
+exoplatform.chat.meetingnotes.task.assignedTo=assigned to
+exoplatform.chat.meetingnotes.task.due=due
+
+exoplatform.chat.meetingnotes.events=Events
+exoplatform.chat.meetingnotes.events.event=Event
+exoplatform.chat.meetingnotes.events.start=Start
+exoplatform.chat.meetingnotes.events.end=EZnd
```
Comment (trangvh) [inline server/src/main/webapp/WEB-INF/classes/locale/chat/server/Resource.properties]: EZnd -> End

**Comment by rdenarie** [comment]
Trang, I update the PR to match your suggestions.

File: server/src/main/java/org/exoplatform/chat/server/ChatServer.java [hunk 7]
```diff
@@ -55,17 +57,16 @@
 import javax.mail.internet.MimeMessage;
 import java.io.IOException;
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.GregorianCalendar;
-import java.util.List;
-import java.util.Properties;
+import java.util.*;
```
Comment (trangvh) [inline server/src/main/java/org/exoplatform/chat/server/ChatServer.java]: It would be better to import specific classes.

**Comment by trangvh** [comment]
Merged to stable/1.0.x.

**Pull request closed without merging**